mission_control/test/test_attitude_servo_behavior.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+"""
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, QinetiQ, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of QinetiQ nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+"""
+import sys
+import os
+import unittest
+import math
+import rosnode
+import rospy
+import rostest
+from mission_control.msg import LoadMission
+from mission_control.msg import ExecuteMission
+from mission_control.msg import ReportExecuteMissionState
+from mission_control.msg import ReportLoadMissionState
+from mission_control.msg import ReportMissions
+from mission_control.msg import AttitudeServo
+from pose_estimator.msg import CorrectedData
+from geometry_msgs.msg import Vector3
+
+
+def wait_for(predicate, period=1):
+    while not rospy.is_shutdown():
+        result = predicate()
+        if result:
+            return result
+        rospy.sleep(period)
+    return predicate()
+
+
+class TestAttitudeServoBehavior(unittest.TestCase):
+    """ 
+        The test loads and execute a attitude servo mission.
+        -   Load the mission attitude_servo_mission_test.xml
+        -   Execute the mission
+        -   Check if the behavior publishes the goal
+        -   Simulate PoseEstimator data to finish the behavior
+        -   Test if the mission is SUCCESS
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        rospy.init_node('test_attitude_servo_behavioral')
+
+    def setUp(self):
```

Comment:
@patapufate meta: could we start reusing test fixtures? Or at least the fraction these share. I see lots of copy and paste here and there.

Reply:
In these behavior test I really want to keep separated.